File: frontend/src/types/access-report.ts
export type PermissionType =
  | "ADMIN"
  | "MAINTAIN"
  | "PUSH"
  | "TRIAGE"
  | "PULL"
  | "UNKNOWN"
  | string;

export interface RepositoryAccess {
  repositoryName: string;
  repositoryFullName: string;
  permission: PermissionType;
}

export interface UserAccess {
  username: string;
  repositories: RepositoryAccess[];
}

export interface AccessReport {
  organization: string;
  generatedAt: string;
  totalRepositories: number;
  totalUsers: number;
  users: UserAccess[];
}

export interface ApiErrorResponse {
  timestamp?: string;
  status?: number;
  error?: string;
  message: string;
}

File: backend/src/main/java/com/ganesh/github_access_report/model/report/AccessReport.java
package com.ganesh.github_access_report.model.report;

import java.time.Instant;
import java.util.List;

public record AccessReport(
        String organization,
        Instant generatedAt,
        int totalRepositories,
        int totalUsers,
        List<UserAccess> users
) {
}
